src/api/utils.go

```diff
@@ -126,6 +127,17 @@ func JsonHandler(f HandlerFunc) errorFunc {
 	}
 }
 
+// Adds CORS headers to a response.
+func CORS(w http.ResponseWriter, methods ...string) {
+	w.Header().Set("Accept", "application/json")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
+	w.Header().Del("Access-Control-Allow-Methods")
+	for _, s := range methods {
+		w.Header().Add("Access-Control-Allow-Methods", s)
```

Comment:
The way golang represents the headers internally is using a multi-map.

When you call `w.Header().Add` it tracks the values added so far in this multimap, and it writes all the values at once when `w.WriteHeader` is called (called implicitly the first time you call `w.Write`).

See the [net/http.Header](https://golang.org/pkg/net/http/#Header) and [net/http.ResponseWriter](https://golang.org/pkg/net/http/#ResponseWriter) for details.